src/query_optimizer/rule_query_optimizer.py

```diff
@@ -0,0 +1,225 @@
+from expression.constant_value_expression import ConstantValueExpression
```

Comment:
Can you please modify the commenting style? Use docstring. I will look at the functionality once you are done with this.

Reply:
Changed them for rule_query_optimizer.py and the abstract plans.